x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/transport/nio/SSLOutboundBuffer.java

```diff
@@ -62,10 +59,7 @@ FlushOperation buildNetworkFlushOperation(BiConsumer<Void, Exception> listener)
             byteBuffers[i] = page.byteBuffer();
```

Comment:
This is where we'd throw with the new assertion if close happened before and now the IO loop would incorrectly think there's still bytes to flush here when those bytes have been released already. It's not really a bug since this would only happen for closed (closed on the same thread) channels anyway so no released bytes would be written just a cosmetic thing.